File: core/src/com/derekentringer/gizmo/component/screen/AbstractScreen.java
package com.derekentringer.gizmo.component.screen;

import com.badlogic.gdx.Screen;
import com.derekentringer.gizmo.Gizmo;

public abstract class AbstractScreen implements Screen {

    protected Gizmo mGizmoGame;

    public AbstractScreen(Gizmo gizmoGame) {
        mGizmoGame = gizmoGame;
    }

    @Override
    public void pause() {
    }

    @Override
    public void resume() {
    }

    @Override
    public void dispose() {
        this.dispose();
    }

}

File: core/src/com/derekentringer/gizmo/component/screen/AbstractScreen.kt
package com.derekentringer.gizmo.component.screen

import com.badlogic.gdx.Screen
import com.badlogic.gdx.assets.AssetManager
import com.derekentringer.gizmo.Gizmo

abstract class AbstractScreen(protected var mGizmoGame: Gizmo) : Screen {

    protected var assetManager = AssetManager()

    override fun pause() {
    }

    override fun resume() {
    }

    override fun dispose() {
        this.dispose()
    }

}